### PR TITLE
chore: Update `dependabot.yml` template

### PR DESCRIPTION
**Pull request by ddixit14**
Dependabot `ignore` feature: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore

### CODE REVIEW

File: synthtool/gcp/templates/java_library/.github/dependabot.yml [hunk 1]
```diff
@@ -11,4 +11,7 @@ updates:
     schedule:
       interval: "daily"
     # Disable version updates for pip dependencies
-    open-pull-requests-limit: 0
```
Comment (suztomo) [inline synthtool/gcp/templates/java_library/.github/dependabot.yml:14]: This already says "# Disable version updates for pip dependencies"

Comment (suztomo) [reply synthtool/gcp/templates/java_library/.github/dependabot.yml:14]: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#open-pull-requests-limit

> This option has no impact on security updates, which have a separate, internal limit of ten open pull requests.

**Comment by alicejli** [comment]
General question : is this update because we're moving to renovate-bot?

**Review by suztomo — APPROVED** [review]

File: synthtool/gcp/templates/java_library/.github/dependabot.yml (hunk 1 above)
Comment (BenWhitehead) [reply synthtool/gcp/templates/java_library/.github/dependabot.yml:14]: I think the security category is how this is leaking through.

* https://github.com/googleapis/java-storage-nio/blob/main/.github/dependabot.yml#L14

Releases notes for the dependency list security in what was fixed.
* https://github.com/googleapis/java-storage-nio/pull/1180

Cryptography probably implicitly gets categorized as security
* https://github.com/googleapis/java-storage-nio/pull/1188

**Comment by suztomo** [comment]
@alicejli  `# we use renovate-bot as well as shared-dependencies BOM to update maven dependencies.`

File: synthtool/gcp/templates/java_library/.github/dependabot.yml (hunk 1 above)
Comment (suztomo) [reply synthtool/gcp/templates/java_library/.github/dependabot.yml:14]: Yes.

**Pull request merged**